lib/qesapdeployment.pm

```diff
@@ -379,7 +393,7 @@ sub qesap_execute {
 sub qesap_prepare_env {
     my (%args) = @_;
     my $variables = $args{openqa_variables};
-    my $provider = lc get_required_var('PUBLIC_CLOUD_PROVIDER');
+    my $provider = $args{provider};
```

Comment:
Maybe we can convert the provider name to lowercase here?
Just to make it more tolerable against incorrect input.

Reply:
Agree, 
Generally speaking, this change is an API change. This PR is mostly for Trento testing, and Trento is not using qesap_prepare_env yet. I decided to revert this change at all.